verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -317,6 +342,59 @@ std::vector<verible::lsp::Location> SymbolTableHandler::FindReferencesLocations(
   return locations;
 }
 
+verible::lsp::Range SymbolTableHandler::FindRenameLocations(
+    const verible::lsp::PrepareRenameParams &params,
+    const verilog::BufferTrackerContainer &parsed_buffers) {
+  Prepare();
+  verible::LineColumnRange symbol =
+      GetTokenRangeAtTextDocumentPosition(params, parsed_buffers);
+  verible::lsp::Range range;
+  range.start.line = symbol.start.line;
+  range.start.character = symbol.start.column;
+  range.end.line = symbol.end.line;
+  range.end.character = symbol.end.column;
+  return range;
```

Comment:
What if we are over a token where a rename would not work ? The protocol definition says that in that case `null` is to be returned.
I guess that means we should return here a `std::optional<verible::lsp::Range>`  - essentially figuring out if the cursor is even hovering over something that can be rennamed (in the first simple implementation we could just look that we're not over a whitespace or keyword token, later we can look if we are actually over a symbol known to the symbol table).

And then in the `textDocument/prepareRename` request handler something that checks if the return `has_value()` and if not a nullptr.